src/components/SongList/index.js

```diff
@@ -0,0 +1,39 @@
+import {
+  mapObjIndexed,
+  prop,
+  reverse,
+  sortBy,
+} from 'ramda'
+import { compose } from 'redux'
+import { connect } from 'react-redux'
+import { firebaseConnect } from 'react-redux-firebase'
+import SongList from './SongList';
+
+const sortDescByVotes = compose(
+  reverse,
+  sortBy(prop('votes')),
+  Object.values,
+  mapObjIndexed((song, id) => ({ ...song, id }))
+)
+
+export default compose(
+  firebaseConnect([
+    'songs'
+  ]),
+  connect(
+    ({ firebase }) => ({
+      songs: sortDescByVotes(firebase.data.songs),
+    }),
+    {},
+    (stateProps, dispatchProps, ownProps) => ({
+      ...stateProps,
+      ...dispatchProps,
+      ...ownProps,
+      changeVote: song => value => {
```

Comment:
ideally, would this function live somewhere else? As we add more actions I could see this getting messy

Reply:
Yes, good point. I’m going to write this as a redux action creator in a subsequent PR.